neo/public/javascripts/map/controllers.js

```diff
@@ -98,7 +105,20 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
         if ($scope.map) {
           $scope.status.zoomLevel = $scope.map.getZoom();
           $scope.bounds = $scope.map.getBounds();
-          if ($scope.status.zoomLevel > 5) {
+          if($scope.status.zoomLevel > 7) {
+            $scope.status.logicLevel = 'city';
+            if (!$scope.status.init) {
+              resetGeoIds($scope.bounds, $scope.geojsonData.city, 'cityID');
+              Asterix.parameters.geoLevel = 'city';
+              Asterix.queryType = 'zoom';
+              Asterix.query(Asterix.parameters, Asterix.queryType);
+            }
+            if ($scope.polygons.statePolygons) {
```

Comment:
since we have three levels now, we should also check the `|| $scope.polygons.countyPolygons` ?